### DISCUSSION TITLE
Netmiko discussions

### DISCUSSION THREAD

**Discussion by ktbyers** [ktbyers/netmiko#2097] [General]
Trying this as a forum for sharing common solutions to issues so that they show up better in search results (and so that GitHub issues doesn't get overloaded with common problems that people run into).

**Comment by AJ-Acevedo** [comment]
It looks like the [Slack invitation](https://join.slack.com/t/pynet/shared_invite/enQtNTA2MDI3NjU0MTM0LTIyZDdhMTBlOWNmNDJhNjkxZTEyMDg3NTRkOWIxMTUwOTAzYmQ0ZjMwMGMyNTM4N2E1YzA3YWQ1MWFiOWM1YzU) link the the README is no longer working. Is that Slack channel still active or are you looking to move the discussion here?

**Reply by ktbyers** [reply]
The Slack invitation is fixed for now. It will eventually break again as Slack doesn't keep their `permanent invites` alive (i.e. they aren't really permanent).

**Reply by Scifideity** [reply]
Just found this post as i was poking around for some guidance. I tried it and looks like it's already broken again.

**Reply by AJ-Acevedo** [reply]
@Scifideity The link on the README is still working.

**Reply by harshita93** [reply]
Hi @ktbyers , It looks like the slack channel invitation link is broken again!.

**Reply by schautwb** [reply]
@harshita93 [https://join.slack.com/t/pynet/shared_invite/zt-2cum7v6nm-MOaMofF1eM_B8~c8NFEvKw](url)

**Reply by jdpfda** [reply]
Can we get a new link?   Looks like the current one and the one above are expired.

**Reply by schautwb** [reply]
@jdpfda [https://join.slack.com/t/pynet/shared_invite/zt-3d55e0ql1-e1NKfSgCEC0tMx6LlCMVUg](url)

**Comment by jatin-rout** [comment]
Hi Folks,
I was trying to make telnet connection to [Connection Technology Systems] switches. But I could not find the compatible device_type. Could you please guide me how can use Netmiko telnet functionality for [Connection Technology Systems] switches. 
Thanks

**Reply by ktbyers** [reply]
@jatin-practice If there is no driver, you or someone would need to create one (or see if there is an existing device_type that is close enough--maybe `cisco_ios` or `generic`)

**Reply by jatin-rout** [reply]
Hi @ktbyers , 
Thanks for quick reply. I tried both cisco_ios and generic. In case of cisco_ios it is failing during login , might be the expecting pattern is not matching and in case of **generic** , problems coming during execution of commands.

**Reply by ktbyers** [reply]
Yep, sounds like you or someone would need to create a driver (or you would need to dig more into why the `generic` was failing).

**Comment by queridiculo** [comment]
Good evening.

I have started looking at netmiko to manage network devices and I have run into a challenge. 

Access to our network equipment is gated by a custom SSH proxy that accepts the destination device as the first argument of a remote command and either allocates a pseudo terminal to that device, or executes a remote command if the application receives more than one argument.

A connection attempt using OpenSSH from the shell would look something like this:

```
ssh -t <ssh.proxy.host> <remote.network.device> <optional arguments/remote commands>
```

I know enough Python to get myself in trouble, but reading through the code so far I haven't quite been able to figure out what would be a sensible approach to use netmiko given this constraint.

Any pointers would be appreciated.

Thanks!

**Reply by ktbyers** [reply]
@queridiculo I would look at this document:

https://pynet.twb-tech.com/blog/netmiko-ssh-proxy-support.html

**Reply by queridiculo** [reply]
Yeah, that's a document I have landed at as well.

Unfortunately it doesn't work for my use case.

I'm dealing with a non-forwarding proxy that opens the SSH connection on behalf of the client. 

Essentially as far as the client SSH connections is concerned, I am always connecting to the proxy itself.

The proxy interprets the first positional argument of the remote SSH command as the remote host it needs to establish a connection to, and then simply returns the stdout to the original client connection.

**Comment by lkristoff** [comment]
Hi @ktbyers ,

I hope this this is the correct place to pose this question.   I'm using Netmiko 4.3.0 and it seems like it is using the wrong command for paging for device type "aruba_os".   I logged the session and it seems to be using "no paging" instead of "no page" (see below).  I recall seeing this fixed in a previous release.   The switches are running ArubaOS-CX so I think the proper device type should be "aruba_os".

I've verified that the correct command is "no page":

switch# no paging
Invalid input: paging
switch# no page
switch# 

Here's an excerpt of the netmiko log:

DEBUG:netmiko:write_channel: b'\r'
DEBUG:netmiko:read_channel:
DEBUG:netmiko:read_channel:
mainlib-3694-esw-1#
DEBUG:netmiko:Pattern found: (mainlib\-3694\-esw\-1.*)
mainlib-3694-esw-1#
DEBUG:netmiko:In disable_paging
DEBUG:netmiko:Command: no paging^M
DEBUG:netmiko:write_channel: b'no paging\r'
DEBUG:netmiko:read_channel:
DEBUG:netmiko:read_channel: no paging
Invalid input: paging

**Reply by lkristoff** [reply]
It looks like there was a bug fix for version 4.0.0 release that said:
-   Change ArubaOS command to disable paging

I assume this was changing it to "no page" but the problem currently exists in at least version 4.3.0.

--Laura

**Reply by ktbyers** [reply]
@lkristoff Unfortunately, there isn't a Netmiko driver for Aruba CX. The device_type `aruba_os` was intended for the Aruba Controllers.

I would probably try a device_type of `aruba_osswitch` (but not sure whether that will work for CX or not).

**Reply by lkristoff** [reply]
Yes that device_type is using "no page".   Seems to work for my purposes so far.   Thanks!
